tilings/tiling.py

```diff
@@ -125,37 +145,38 @@ def from_perms(
             t = t.add_list_requirement(req_list)
         return t
 
+    @cssmethodtimer("Tiling._fill_empty")
     def _fill_empty(self) -> None:
         """Add size one obstructions to cells that are empty."""
         add = []
         (i, j) = self.dimensions
+        active_cells = self.active_cells
+        empty_cells = self.empty_cells
         for x in range(i):
             for y in range(j):
-                if (x, y) not in self.active_cells and (x, y) not in self.empty_cells:
+                if (x, y) not in active_cells and (x, y) not in empty_cells:
```

Comment:
I would replace this big nested statement by
```python
all_cells = set(itertools.product(*map(range, dimensions)))
add = {GriddedPerm.single_cell(Perm((0,)), cell) for cell in all_cells - active_cells - empty_cells}
```
but maybe I'm just obsessed about avoiding big nesting
```

Reply:
Do you claim your version is faster? I find it definitely less readable, but maybe that's worth it if there's a speed advantage.

Reply:
Not claiming that it's faster only that it is more readable. Probably not worth changing if you don't find it more readable